rsmtool/utils/conversion.py

```diff
@@ -43,3 +45,40 @@ def convert_to_float(value):
         Value converted to float, if possible
     """
     return int_to_float(string_to_number(value))
+
+
+def parse_range(value):
+    """
+    Parse range strings used in rsmexplain.
+
+    Valid range strings are two positive integers separated by a hyphen ("10-20").
+
+    Parameters
+    ----------
+    value : str
+        The range string we want to parse.
+
+    Returns
+    -------
+    Tuple[int, int]
+        A tuple containing two integers.
+    """
+    valid_range = False
+    if matchobj := RSMEXPLAIN_RANGE_REGEXP.match(value):
+        groups = matchobj.groupdict()
+        start_index = string_to_number(groups.get("start"))
+        end_index = string_to_number(groups.get("end"))
+
+        # the range is valid if and only if:
+        # - both start and end indices are positive integers
+        # - and start index < end index
+        valid_range = (
+            isinstance(start_index, int)
+            and isinstance(end_index, int)
+            and 0 <= start_index < end_index
```

Comment:
Should we add an upper bound?

Reply:
if this does what I think it does, it probably makes more sense to evaluate whether the range exceeds the length of your feature set at the subsampling step. If your upper bound exceeds the total length of the set, just re-evaluate it to len(feature_set) and send a log message indicating that the range was larger than the set.
Otherwise you need to set an arbitrary upper limit or check the feature set length at this step here.

Reply:
We don't need an upper bound because even the endpoint is larger, the Python indexing will work just fine without raising an error. For example.

```
In [1]: a = list(range(1, 100))

In [2]: a[90:110]
Out[2]: [91, 92, 93, 94, 95, 96, 97, 98, 99]
```

Reply:
We should try and add a test for this to be sure.

Reply:
Sure. I will add it.